### PR TITLE
Update data parse

### PR DESCRIPTION
**Pull request by secant**
Currently, this will write all the topics, questions, and answers into the database.

TODO:
- ~~Write dependencies to Question table~~
- Query for parent topic object

### CODE REVIEW

File: load_data.py [hunk 1]
```diff
@@ -10,31 +10,37 @@
 from django.core.management.sql import sql_delete
 from django.db import connections, DEFAULT_DB_ALIAS, models
 from django.db.utils import IntegrityError
```
Comment (secant) [inline load_data.py]: Will remove, using ValidationError instead

File: load_data.py [hunk 2]
```diff
@@ -10,31 +10,37 @@
...
-    schema_obj = AnalysisType(
+    schema_parent = schema['parent']
+    # if schema_parent:
+    #     p_id = Topic.objects.get(name=schema_parent).id #TODO: Query for parent.
+    # else:
+    p_id = None
+    schema_obj = Topic(
+        parent = p_id,
         name=schema_name,
-        requires_processing=schema_name not in ['Useless', 'Future'],
```
Comment (secant) [inline load_data.py]: @nickbadams @FadyShoukry 
What is the purpose of `requires_processing` in `analysis_type` (now topic)?

File: parse_schema.py [hunk 3]
```diff
@@ -13,35 +14,54 @@ class TopicsSchemaParser(object):
     """
     Parses a json schema of topics and questions and populates the database
     """
-    def __init__(self, schema, analysis_type):
+    def __init__(self, topic_obj, schema, dependencies):
         """
         schema: a json schema as a string or loaded json
```
Comment (secant) [inline parse_schema.py]: Update

File: parse_schema.py [hunk 4]
```diff
@@ -13,35 +14,54 @@ class TopicsSchemaParser(object):
...
+    def clean_dependencies(self):
+        """
+        Returns a list of named tuples that represent each dependency:
+        [Dependency(topic, question, answer, next_topic, next_topic)]
+        Also converts strings to integers.
+        """
+        Dependency = namedtuple('Dependency', ['topic', 'question', 'answer', 'next_topic', 'next_question'])
+        clean_dep = []
+        for dep in self.dep:
+            new_dep = dep[0].split(".")
```
Comment (secant) [inline parse_schema.py]: use source, target = dep to unpack

**Comment by stefanv** [comment]
@secant Will you please push the reworked version?

**Comment by secant** [comment]
Merged w/ master under new_model_refactor.

**Pull request closed without merging**